ArcGISRuntimeSDKQt_CppSamples/EditData/SnapGeometryEdits/SnapGeometryEdits.cpp

```diff
@@ -0,0 +1,335 @@
+// [WriteFile Name=SnapGeometryEdits, Category=EditData]
+// [Legal]
+// Copyright 2024 Esri.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// [Legal]
+
+#ifdef PCH_BUILD
+#include "pch.hpp"
+#endif // PCH_BUILD
+
+#include "SnapGeometryEdits.h"
+
+#include "FeatureLayer.h"
+#include "FeatureTable.h"
+#include "Geometry.h"
+#include "GeometryEditor.h"
+#include "GeometryEditorElement.h"
+#include "GeometryTypes.h"
+#include "Graphic.h"
+#include "GraphicListModel.h"
+#include "GraphicsOverlay.h"
+#include "GraphicsOverlayListModel.h"
+#include "IdentifyGraphicsOverlayResult.h"
+#include "Layer.h"
+#include "LayerListModel.h"
+#include "LoadSettings.h"
+#include "Map.h"
+#include "MapQuickView.h"
+#include "MapTypes.h"
+#include "Portal.h"
+#include "PortalItem.h"
+#include "SimpleFillSymbol.h"
+#include "SimpleLineSymbol.h"
+#include "SimpleMarkerSymbol.h"
+#include "SnapGeometryEdits.h"
+#include "SnapSettings.h"
+#include "SnapSourceSettings.h"
+#include "SymbolTypes.h"
+
+#include <QFuture>
+
+using namespace Esri::ArcGISRuntime;
+
+SnapGeometryEdits::SnapGeometryEdits(QObject* parent /* = nullptr */) :
+  QObject(parent)
+{
+  Portal* portal = new Portal(this);
+  PortalItem* portalItem = new PortalItem(portal, "b95fe18073bc4f7788f0375af2bb445e", this);
+  m_map = new Map(portalItem, this);
+
+  m_geometryEditor = new GeometryEditor(this);
+  m_graphicsOverlay = new GraphicsOverlay(this);
+  connect (m_map, &Map::doneLoading, this, [this]()
+  {
+    for (Layer* layer : *m_map->operationalLayers())
+    {
+      connect (layer, &Layer::doneLoading, this, [this]()
+      {
+        // Enable snap settings after layers load
+        if (m_layersLoaded == false)
+        {
+          m_layersLoaded = true;
```

Comment:
Can you explain this logic?

Reply:
used to enable the snap settings button after layers are loaded